frontend/src/components/Participants.vue

```diff
@@ -0,0 +1,288 @@
+        transition="slide-y-transition"
+        >{{ getError }}</v-alert
+      >
+      <v-alert
+        type="warning"
+        :value="warningAlert"
+        v-model="warningAlert"
+        dismissible
+        @click="warningAlert = !warningAlert"
+        transition="slide-y-transition"
+        >{{ warningMessage }}</v-alert
+      >
+      <v-alert
+        type="success"
+        :value="successAlert"
+        v-model="successAlert"
+        dismissible
+        @click="successAlert = !successAlert"
+        transition="slide-y-transition"
+        >Participant has been successfully added</v-alert
+      >
+    </v-flex>
+    <v-card>
+      <v-toolbar flat>
+        <v-icon class="pa-3">fas fa-user-plus</v-icon>
+        <v-toolbar-title>Create new participant</v-toolbar-title>
+      </v-toolbar>
+      <v-divider></v-divider>
+      <v-form @keyup.enter="onSubmit" class="pa-6" ref="form">
+        <v-text-field
+          type="text"
+          label="E-mail"
+          v-model.trim="form.email"
+          @blur="$v.form.email.$touch()"
+          :error-messages="emailErrors"
+        />
+        <v-text-field
+          label="Github link"
+          v-model="form.github"
+          @blur="$v.form.github.$touch()"
+          :error-messages="githubErrors"
+        />
+        <v-text-field
+          label="First name"
+          v-model="form.first_name"
+          @blur="$v.form.first_name.$touch()"
+          :error-messages="firstNameErrors"
+        />
+        <v-text-field
+          label="Last name"
+          v-model="form.last_name"
+          @blur="validateLastName($event)"
+          :error-messages="lastNameErrors"
+        />
+        <v-text-field
+          label="Slack nick"
+          v-model="form.slack"
+          @blur="validateSlack($event)"
+          :error-messages="slackErrors"
+        />
+        <v-text-field
+          label="Phone No."
+          v-model="form.phone"
+          :counter="9"
+          @blur="validatePhone($event)"
+          :error-messages="phoneErrors"
+        />
+        <v-card-actions class="pt-3 align-center justify-center">
+          <v-btn
+            :disabled="$v.$invalid"
+            @click="onSubmit"
+            class="add-participant-btn align-center justify-center"
+            >Add new participant</v-btn
+          >
+        </v-card-actions>
+      </v-form>
+    </v-card>
+  </v-container>
+</template>
+
+<script>
+import { mapState } from 'vuex';
+import { mapGetters } from 'vuex';
+import {
+  required,
+  minLength,
+  maxLength,
+  email,
+  numeric
+} from 'vuelidate/lib/validators';
+export default {
+  data() {
+    return {
+      form: {
+        email: '',
+        github: '',
+        first_name: '',
+        last_name: '',
+        slack: '',
+        phone: ''
+      },
+      warningAlert: false,
+      warningMessage: '',
+      successAlert: false
+    };
+  },
+  created() {
+    this.$store.dispatch('participant/getParticipants');
```

Comment:
We are dispatching this action twice: when creating Participants component and HacknightsParticipants component. Since both are child components for Dashboard, maybe it is worth to move that code level up - to dashboard?

Reply:
Agreed, but since both would be the children of dashboard should we pass it somehow? It's the unknown territory for me.

Reply:
There is one store for whole app and You would be able to access participants with `this.$store.dispatch('participant/getParticipants');` same as now. With `this.$store.dispatch('participant/getParticipants')` we pulling participants from backend and "storing" them in central place that every component has access to these same set of data.
So right now we send request to backend when creating Participants component and second time when creating HacknightsParticipants component.

Reply:
Done